mvc/overview.rst

```diff
@@ -1,8 +1,210 @@
-.. include:: /../common/stub-topic.txt
+Overview of ASP.NET MVC
+=======================
+By `Tom Archer`_, `Daniel Roth`_
 
-|stub-icon| Overview of ASP.NET MVC
-===================================
+.. image:: overview/_static/mvc.png
+  :align: right
 
-.. include:: /../common/stub-notice.txt
+ASP.NET MVC is a lightweight, cross-platform, highly testable Web application development framework that separates an application into three main components: model, view, and controller:
 
-.. _issue: https://github.com/aspnet/Docs/issues/113
+- :doc:`Models </models/index>` are the parts of the application that implement the logic for the application's data domain. Often, model objects retrieve and store model state in a database. For example, a Product object might retrieve information from a database, operate on it, and then write updated information back to a Products table in a SQL Server database.
+
+- :doc:`Views </views/index>` are responsible for displaying the application's user interface (UI). Typically, this UI is created from the model data. An example would be an edit view of a Products table that displays text boxes, drop-down lists, and check boxes based on the current state of the Product object being represented in the view.
+
+- :doc:`Controllers </controllers/index>` process incoming requests, work with the model, and ultimately return a view to the client that displays the user interface.
+
+This delineation of responsibilities helps you scale the application in terms of complexity because it’s much easier to code, debug, and test something (model, view, or controller) that has a single job rather than something that performs multiple tasks.
+
+Features
+--------
+
+ASP.NET MVC provides the following features:
+  - `Routing`_
+  - `Model binding`_
+  - `Request validation`_
+  - `Razor view engine`_
+  - `Strongly typed views`_
+  - `HTML Helpers`_
+  - `Tag Helpers`_
+  - `View Components`_
+  - `Dependency Injection`_
+  - `Filters`_
+  - `Areas`_
+  - `Mobile development`_
+  - `Web UI and Web APIs`_
+  - `Test-driven Development`_
+  - `Built on top of ASP.NET 5`_
+
```

Comment:
I don't really think of extensibility as a feature - it's more a result of good framework design.

Regarding security, I do mention authorization support in the filters section. Authentication and membership are mentioned in the Built on top of ASP.NET 5 section. I can also mention CORS in the Web API section.

I do talk about integration with jQuery validation in the request validation section. Intellisense support is mentioned in the strongly typed models section, but it might be work adding a top level feature for tooling (scaffolding, IntelliSense, item and project templates, authentication configuration)